compiler/noirc_evaluator/src/ssa/interpreter/tests/instructions.rs

```diff
@@ -0,0 +1,840 @@
+use std::sync::Arc;
+
+use iter_extended::vecmap;
+use noirc_frontend::Shared;
+
+use crate::ssa::{
+    interpreter::{
+        NumericValue, Value,
+        tests::{expect_value, expect_values},
+        value::ReferenceValue,
+    },
+    ir::{
+        types::{NumericType, Type},
+        value::ValueId,
+    },
+};
+
+use super::{executes_with_no_errors, expect_error};
+
+#[test]
+fn add() {
+    let value = expect_value(
+        "
+        acir(inline) fn main f0 {
+          b0():
+            v0 = add i32 2, i32 100
+            return v0
+        }
+    ",
+    );
+    assert_eq!(value, Value::Numeric(NumericValue::I32(102)));
+}
+
+/// TODO: Replace panic with error
+#[test]
+#[should_panic(expected = "called `Option::unwrap()` on a `None` value")]
+fn add_overflow() {
+    expect_error(
+        "
+        acir(inline) fn main f0 {
+          b0():
+            v0 = add u8 200, u8 100
+            return v0
+        }
+    ",
+    );
+}
+
+#[test]
+fn add_unchecked() {
+    executes_with_no_errors(
+        "
+        acir(inline) fn main f0 {
+          b0():
+            v0 = unchecked_add u8 200, u8 100
+            return v0
+        }
+    ",
+    );
+}
+
+#[test]
+fn sub() {
+    let value = expect_value(
+        "
+        acir(inline) fn main f0 {
+          b0():
+            v0 = sub i32 10101, i32 101
+            return v0
+        }
+    ",
+    );
+    assert_eq!(value, Value::Numeric(NumericValue::I32(10000)));
+}
+
+/// TODO: Replace panic with error
+#[test]
+#[should_panic(expected = "called `Option::unwrap()` on a `None` value")]
+fn sub_underflow() {
+    executes_with_no_errors(
```

Comment:
This is a bit confusing. Maybe use `expect_error` instead, so the `should_panic` and `executes_with_no_errors` don't contradict each other? Same for the other tests where `should_panic` is used below.

Reply:
`expect_error` won't work on its own currently since we panic instead of returning an error. I intended this to be temporary until https://github.com/noir-lang/noir/issues/8284 but should be able to change `expect_error` to use `catch_unwind` in the meantime. I'll update it.

Reply:
Unfortunately, we can't change this currently. I tried adding a catch_unwind to expect_error but `Ssa` contains a frontend Type in its error selector field which makes it unsafe to pass a catch unwind boundary so Rust errors.
For now we'll need to keep the `should_panic` versions and when error variants are added we can move to `expect_error`

Reply:
Actually all I really meant was whether it would work like this:
```rust
[should_panic(expected = "called `Option::unwrap()` on a `None` value")]
fn sub_underflow() {
    expect_error(
```
that is, which is just a nit to say that "should panic" contradicts "expect no errors" in spirit, and I thought "expect error" could communicate better that in the future, when `should_panic` is removed, we still expect an error, but a runtime one, not a panic.

Reply:
Oh, I was thinking the opposite. That "should_panic" would contradict "expect error" because it'd imply you're expecting an error, didn't find one, and panicked.

Reply:
Ah okay, never mind then 😄

Reply:
Ah - just updated this anyway :man_shrugging: